### ISSUE TITLE
continuous builds for build tools that we consume

### ISSUE THREAD

**Issue by dustymabe** [coreos/fedora-coreos-tracker#84]
In our current state of heavy development we have a need to iterate fast on our build tools for Fedora CoreOS. In the past we have been achieving this goal by setting up [rdgo](https://github.com/projectatomic/rpmdistro-gitoverlay) in centos CI. The problem is that the current setup does not support multi-arch.

We need to find a solution that includes multi-arch but also gives us fast access to builds. Options include:

- somehow setting up rdgo to support multi-arch (multi-arch in centos ci?)
- use copr and automate building
- request a new koji tag and automated building + repo generation

The build tools we would currently set this up for include

- ostree
- rpm-ostree

**Comment by dustymabe** [comment]
>    request a new koji tag and automated building + repo generation

This option is the one that I prefer. If no one objects I can start conversations to see if this is feasible.

**Comment by sinnykumari** [comment]
> > request a new koji tag and automated building + repo generation
> 
> This option is the one that I prefer. If no one objects I can start conversations to see if this is feasible.

+1

**Comment by jlebon** [comment]
I guess I'm just trying to understand how it'd work. Would this essentially be like `fedpkg push && fedpkg build` or `fedpkg build --srpm`?

**Comment by dustymabe** [comment]
> I guess I'm just trying to understand how it'd work. Would this essentially be like `fedpkg push && fedpkg build` or `fedpkg build --srpm`?

My thoughts are that we would have a separate branch in dist-git that we update with bots (i.e. automated). SCM syncing, Builds, tagging, repo generation would be automatic.

**Comment by LorbusChris** [comment]
I built rpm-/ostree rpms from master in a copr repo recently: https://copr.fedorainfracloud.org/coprs/lorbus/rpm-ostree/packages/

I had to change both spec files to accomodate this (see https://github.com/LorbusChris/ostree-spec and https://github.com/LorbusChris/rpm-ostree-spec) . However, it may still be helpful as a reference in case we choose to use COPR for our continuous builds.

**Comment by LorbusChris** [comment]
Note so self: https://fedoraproject.org/wiki/Forge-hosted_projects_packaging_automation#Packaging_a_release_.28except_for_GitLab.29

**Comment by cgwalters** [comment]
I would scope this to a much larger set of things.  It's not just build tools it's also things we ship in the host.  For example we clearly want https://github.com/coreos/ignition from master and for that matter https://github.com/coreos/ignition-dracut/ etc etc.

And I see no reason we shouldn't e.g. have builds that track systemd git master and for that matter the kernel.    And NetworkManager, podman.

**Comment by dustymabe** [comment]
> I would scope this to a much larger set of things. It's not just build tools it's also things we ship in the host.

I 100% agree so that we can have a more continuous build (and CI) of FCOS itself. I see the same mechanism we set up for the build tools being able to be used for this purpose.

**Comment by dustymabe** [comment]
Opened a ticket for releng to create us a tag and dist-repo (automatically generated based on rpms tagged into koji tag) for us to experiment with: https://pagure.io/releng/issue/8165

**Comment by LorbusChris** [comment]
Probably orthogonal to the Koji tags:
Source-Git aka [Packit](https://github.com/packit-service/packit) for rpm automation. Tomas said he hopes a first usable version will be ready in ~2 weeks. I'll definitely use that for greenboot.

**Comment by dustymabe** [comment]
> Probably orthogonal to the Koji tags:
> Source-Git aka [Packit](https://github.com/packit-service/packit) for rpm automation. Tomas said he hopes a first usable version will be ready in ~2 weeks. I'll definitely use that for greenboot.

yep. I actually gave the documentation a read through yesterday and watched the video from devconf. I'm excited to learn more and give feedback.

**Comment by cgwalters** [comment]
I am still a bit skeptical about packit.  But anyways just a note, I did create https://copr.fedorainfracloud.org/coprs/g/CoreOS/continuous/ and kind of try to maintain it here and there.
